consumer/src/mode/decoded/deposited_curve.rs

```diff
@@ -0,0 +1,333 @@
+
+        Ok(())
+    }
+
+    /// This function creates a `Signal` for the `DepositedCurve` event
+    async fn create_signal(
+        &self,
+        decoded_consumer_context: &DecodedConsumerContext,
+        event: &DecodedMessage,
+        curve_vault: &CurveVault,
+    ) -> Result<(), ConsumerError> {
+        if self.senderAssetsAfterTotalFees > U256::from(0) {
+            if let Some(atom_id) = curve_vault.atom_id.clone() {
+                Signal::builder()
+                    .id(DecodedMessage::event_id(event))
+                    .account_id(self.sender.to_string().to_lowercase())
+                    .delta(U256Wrapper::from(self.senderAssetsAfterTotalFees))
+                    .atom_id(atom_id)
+                    .deposit_id(DecodedMessage::event_id(event))
+                    .block_number(U256Wrapper::try_from(event.block_number)?)
+                    .block_timestamp(event.block_timestamp)
+                    .transaction_hash(event.transaction_hash.clone())
+                    .build()
+                    .upsert(
+                        &decoded_consumer_context.pg_pool,
+                        &decoded_consumer_context.backend_schema,
+                    )
+                    .await?;
+            } else if let Some(triple_id) = curve_vault.triple_id.clone() {
+                Signal::builder()
+                    .id(DecodedMessage::event_id(event))
+                    .account_id(self.sender.to_string().to_lowercase())
+                    .delta(U256Wrapper::from(self.senderAssetsAfterTotalFees))
+                    .triple_id(triple_id)
+                    .deposit_id(DecodedMessage::event_id(event))
+                    .block_number(U256Wrapper::try_from(event.block_number)?)
+                    .block_timestamp(event.block_timestamp)
+                    .transaction_hash(event.transaction_hash.clone())
+                    .build()
+                    .upsert(
+                        &decoded_consumer_context.pg_pool,
+                        &decoded_consumer_context.backend_schema,
+                    )
+                    .await?;
+            }
+        }
+        Ok(())
+    }
+
+    /// This function creates a `Deposit` record for the `DepositedCurve` event
+    async fn create_deposit(
+        &self,
+        decoded_consumer_context: &DecodedConsumerContext,
+        event: &DecodedMessage,
+    ) -> Result<String, ConsumerError> {
+        // Create the deposit record
+        let deposit_id = DecodedMessage::event_id(event);
+        
+        // Use sqlx to insert the deposit record directly
+        // Cast string values to numeric using PostgreSQL's CAST function
+        sqlx::query(&format!(
+            r#"
+            INSERT INTO {}.deposit 
+                (id, sender_id, receiver_id, receiver_total_shares_in_vault, sender_assets_after_total_fees, shares_for_receiver, entry_fee, vault_id, is_triple, is_atom_wallet, block_number, block_timestamp, transaction_hash) 
```

Comment:
I want to be sure this is writing to the curve vault database entry, not the corresponding base vault

Reply:
We should use the wrappers for these instead of doing raw SQL lol (my bad)